frontend/src/hooks/useCustomEvent.ts

```diff
@@ -8,10 +8,10 @@ export const useCustomEvent = (
     eventName: string,
     handler: (event: CustomEvent) => void,
 ) => {
-    const emitEvent = () => {
+    const emitEvent = useCallback(() => {
```

Comment:
give stable identity to this function so when it's used in useEffect deps it doesn't change